tests/package_pkg_resources/main.py

```diff
@@ -19,26 +19,94 @@
 """
 
 
+import atexit
+import os
+import shutil
+import tempfile
+
+
 def main():
     print('In pkg_resources test main()')
     try:
         import pkg_resources
+        import setuptools
+        # We can't query pkg_resources' version so query setuptools instead
+        version = pkg_resources.parse_version(setuptools.__version__)
+        minimum = pkg_resources.parse_version('36.6.0')
+        if (version < minimum):
+            print('Skipping test, pkg_resources module is too old')
+            return
     except ImportError:
         print('Skipping test, pkg_resources module is not available')
         return
 
     ws = pkg_resources.working_set
 
+    # Set a safe extraction dir (the default is unsafe)
+    extraction_tmpdir = tempfile.mkdtemp()
+    atexit.register(lambda: shutil.rmtree(
+        extraction_tmpdir, ignore_errors=True))
+    pkg_resources.set_extraction_path(extraction_tmpdir)
+
     # Informational for debugging
     distributions = list(ws)
-    print('Resources found: %s' % distributions)
-
-    # Check for the packages we provided metadata for.  There will
-    # also be metadata for whatever other packages happen to be
-    # installed in the current Python interpreter.
-    for spec in ['portpicker==1.2.0', 'yapf==0.19.0']:
-        dist = ws.find(pkg_resources.Requirement.parse(spec))
-        assert dist, (spec, distributions)
+    assert distributions
+
+    # Test package that doesn't exist.
+    # I hereby promise never to release a package with this name.
+    nonexistent_name = 'subpar-package-does-not-exist-blorg'
+    req = pkg_resources.Requirement.parse(nonexistent_name)
+    dist = ws.find(req)
+    assert not dist
+
+    # Package exists, has code at the top level directory
+    portpicker_spec = 'portpicker==1.2.0'
+    req = pkg_resources.Requirement.parse(portpicker_spec)
+    # Extract single file
+    fn = pkg_resources.resource_filename(req, 'data_file.txt')
+    with open(fn) as f:
+        actual = f.read()
+        assert actual == 'Dummy data file for portpicker\n', actual
+    # Extract all
+    dirname = pkg_resources.resource_filename(req, '')
+    expected = [
+        # The __init__.py file shouldn't be here, but is, as an
+        # unfortunately side effect of Bazel runfiles behavior.
```

Comment:
I think you opened a GitHub issue, do you want to link to it here?

Reply:
Done